### ISSUE TITLE
Rewrite Search Box

### ISSUE THREAD

**Issue by therungg** [therungg/therun-frontend#4]
The search box was one of the first things I've built, and it has *not* aged well. It could use a complete redesign and refactor.

- Better search layout
- Better responsiveness of the search box (more mobile friendly)
- Only search after the user has not input for n milliseconds

**Comment by florence-wolfe** [comment]
Keyboard navigation in the search currently doesn't work as intended. Adding this to the list!!
